docs/proposals/resource-metrics-api.md

```diff
@@ -0,0 +1,177 @@
+it will be probably possible to provide a reasonable implementation of the feature anyway.
+
+#### Kubernetes dashboard
+
+[Kubernetes dashboard](https://github.com/kubernetes/dashboard) in order to draw graps requires resource usage
+in timeseries format from relatively long period of time. The aggreations should be also possible on various levels
+including replication controllers, deployments, services, etc.
+
+Since the use case is complicated it will not be supported initally in the API and they will query Heapster
+directly using some custom API there.
+
+## Proposed API
+
+Initially the metrics API will be in a separate [API group](api-group.md) called ```metrics```.
+Later if we decided to have Node and Pod in different API groups also
+NodeMetrics and PodMetrics should be in different API groups.
+
+#### Schema
+
+The proposed schema is as follow. Each top-level object has `TypeMeta` and `ObjectMeta` fields
+to be compatible with Kubernetes API standards.
+
+```go
+type NodeMetrics struct {
+  unversioned.TypeMeta
+  ObjectMeta
+
+  // The following fields define time interval from which metrics were
+  // collected in the following format [Timestamp-Window, Timestamp].
+  Timestamp unversioned.Time
+  Window    unversioned.Duration
+
+  Usage v1.ResourceList
+}
+
+type PodMetrics struct {
+  unversioned.TypeMeta
+  ObjectMeta
+
+  // The following fields define time interval from which metrics were
+  // collected in the following format [Timestamp-Window, Timestamp].
+  Timestamp unversioned.Time
+  Window    unversioned.Duration
+
+  Containers []ContainerMetrics
+}
+
+type ContainerMetrics struct {
+  Name  string
+  Usage v1.ResourceList
+}
+```
+
+By default `Usage` is the mean from samples collected within the returned time window.
+The default time window is 1 minute.
+In the future we may want to introduce the following params:
+`aggreator` (`max`, `min`, `95th`, etc.) and `window` (`1h`, `1d`, `1w`, etc.)
+which will allow to get the other aggregates over the custom time window.
+
+#### Endpoints
+
+All endpoints are GET endpoints, rooted at `/apis/metrics/v1alpha1/`.
+There won't be support for the other REST methods.
+
+The list of supported endpoints:
+- `/nodes` - all node metrics; type `[]NodeMetrics`
+- `/nodes/{node}` - metrics for a specified node; type `NodeMetrics`
+- `/namespaces/{namespace}/pods` - all pod metrics within namespace with support for `all-namespaces`; type `[]PodMetrics`
+- `/namespaces/{namespace}/pods/{pod}` - metrics for a specified pod; type `PodMetrics`
+
+The following query parameters are supported:
+- `labelSelector` - restrict the list of returned objects by labels (list endpoints only)
+
+## Further improvements
+
+Depending on the further requirements the following features may be added:
+- support for more metrics
+- support for application level metrics
+- watch for metrics
+- support for different window sizes and aggreation functions
```

Comment:
You can't add multiple windows nicely, since Window is top-level value.   If this is important, then let's design it properly now.  Otherwise, remove mention of multiple window sizes.